extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -133,7 +136,17 @@ void buildImperative(CombinedIndexBuildItem index,
             BuildProducer<BytecodeTransformerBuildItem> transformers,
             BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             BuildProducer<PropertyMappingClassBuildStep> propertyMappingClass,
-            BuildProducer<PanacheEntityClassesBuildItem> entityClasses) {
+            BuildProducer<PanacheEntityClassesBuildItem> entityClasses) throws BuildException {
+
+        // we verify that no ID fields are defined (via @BsonId) when extending PanacheMongoEntity
+        for (AnnotationInstance annotationInstance : index.getIndex().getAnnotations(DOTNAME_BSON_ID)) {
+            ClassInfo info = io.quarkus.panache.common.deployment.JandexUtil.getEnclosingClass(annotationInstance);
+            if (info.superName().equals(DOTNAME_PANACHE_ENTITY)) {
```

Comment:
Same remark about inheritance.

Reply:
Done, I climb up the hierarchy of classes.